.buildkite/scripts/post_github_comment.sh

```
@@ -0,0 +1,177 @@
+check_required_env "BUILDKITE_COMMIT"
+check_required_env "BUILDKITE_REPO"
+check_required_env "BUILDKITE_BRANCH"
+
+# Function to generate JWT token for GitHub App authentication
+generate_jwt_token() {
+  local app_id="$1"
+  local private_key="$2"
+
+  # JWT header
+  local header='{"typ":"JWT","alg":"RS256"}'
+  local header_b64=$(echo -n "$header" | base64 | tr -d '=' | tr '/+' '_-' | tr -d '\n')
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)